### PR TITLE
Database

### PR DESCRIPTION
**Pull request by MilosKozak**

### CODE REVIEW

File: app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.java [hunk 1]
```diff
@@ -638,59 +640,47 @@ public void run() {
...
-                    .duration(defHelper.determineActivityTTDuration())
-                    .reason(MainApp.gs(R.string.activity))
-                    .source(Source.USER)
-                    .low(Profile.toMgdl(target, profile.getUnits()))
-                    .high(Profile.toMgdl(target, profile.getUnits()));
-            TreatmentsPlugin.getPlugin().addToHistoryTempTarget(tempTarget);
+            BlockingAppRepository.INSTANCE.runTransaction(new InsertTemporaryTargetAndCancelCurrentTransaction(
+                    System.currentTimeMillis(),
+                    defHelper.determineActivityTTDuration() * 60000,
+                    TemporaryTarget.Reason.EATING_SOON,
```
Comment (AdrianLxM) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.java]: This should be `ACTIVITY`.

File: app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.java [hunk 2]
```diff
@@ -638,59 +640,47 @@ public void run() {
...
-                    .duration(defHelper.determineHypoTTDuration())
-                    .reason(MainApp.gs(R.string.hypo))
-                    .source(Source.USER)
-                    .low(Profile.toMgdl(target, profile.getUnits()))
-                    .high(Profile.toMgdl(target, profile.getUnits()));
-            TreatmentsPlugin.getPlugin().addToHistoryTempTarget(tempTarget);
+            BlockingAppRepository.INSTANCE.runTransaction(new InsertTemporaryTargetAndCancelCurrentTransaction(
+                    System.currentTimeMillis(),
+                    defHelper.determineHypoTTDuration() * 60000,
+                    TemporaryTarget.Reason.EATING_SOON,
```
Comment (AdrianLxM) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.java]: This should be `HYPOGLYCEMIA`.

File: app/src/main/java/info/nightscout/androidaps/plugins/general/overview/dialogs/NewCarbsDialog.java [hunk 3]
```diff
@@ -395,32 +395,26 @@ private void submit() {
...
-                                    .duration(finalActivityTTDuration)
-                                    .reason(MainApp.gs(R.string.activity))
-                                    .source(Source.USER)
-                                    .low(Profile.toMgdl(finalActivityTT, currentProfile.getUnits()))
-                                    .high(Profile.toMgdl(finalActivityTT, currentProfile.getUnits()));
-                            TreatmentsPlugin.getPlugin().addToHistoryTempTarget(tempTarget);
+                            BlockingAppRepository.INSTANCE.runTransaction(new InsertTemporaryTargetAndCancelCurrentTransaction(
+                                    System.currentTimeMillis(),
+                                    finalActivityTTDuration * 60000,
+                                    TemporaryTarget.Reason.EATING_SOON,
```
Comment (AdrianLxM) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/overview/dialogs/NewCarbsDialog.java]: This should be `ACTIVITY`.

File: app/src/main/java/info/nightscout/androidaps/plugins/general/overview/dialogs/NewCarbsDialog.java [hunk 4]
```diff
@@ -395,32 +395,26 @@ private void submit() {
...
-                                    .duration(finalHypoTTDuration)
-                                    .reason(MainApp.gs(R.string.hypo))
-                                    .source(Source.USER)
-                                    .low(Profile.toMgdl(finalHypoTT, currentProfile.getUnits()))
-                                    .high(Profile.toMgdl(finalHypoTT, currentProfile.getUnits()));
-                            TreatmentsPlugin.getPlugin().addToHistoryTempTarget(tempTarget);
+                            BlockingAppRepository.INSTANCE.runTransaction(new InsertTemporaryTargetAndCancelCurrentTransaction(
+                                    System.currentTimeMillis(),
+                                    finalHypoTTDuration * 60000,
+                                    TemporaryTarget.Reason.EATING_SOON,
```
Comment (AdrianLxM) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/overview/dialogs/NewCarbsDialog.java]: This should be `HYPOGLYCEMIA`.

File: app/src/main/java/info/nightscout/androidaps/plugins/general/wear/ActionStringHandler.java [hunk 5]
```diff
@@ -213,7 +219,7 @@ public synchronized static void handleInitiate(String actionstring) {
                 return;
             }
 
-            BgReading bgReading = DatabaseHelper.actualBg();
+            GlucoseValue bgReading = BlockingAppRepository.INSTANCE.getLastRecentGlucoseValue();
```
Comment (AdrianLxM) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/wear/ActionStringHandler.java]: Let's find a better name :)

File: app/src/main/java/info/nightscout/androidaps/plugins/general/wear/ActionStringHandler.java [hunk 6]
```diff
@@ -693,17 +698,19 @@ private static void setCPP(int timeshift, int percentage) {
...
-                .date(System.currentTimeMillis())
-                .duration(duration)
-                .reason("WearPlugin")
-                .source(Source.USER);
-        if (tempTarget.durationInMinutes != 0) {
-            tempTarget.low(low).high(high);
+        if (duration == 0) {
+            try {
+                BlockingAppRepository.INSTANCE.runTransaction(new CancelTemporaryTargetTransaction());
+            } catch (IllegalStateException ignored) {
```
Comment (AdrianLxM) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/wear/ActionStringHandler.java]: I'd log these situations.

Comment (MilosKozak) [reply app/src/main/java/info/nightscout/androidaps/plugins/general/wear/ActionStringHandler.java]: it's on more places

File: app/src/main/java/info/nightscout/androidaps/plugins/source/SourceDexcomPlugin.kt [hunk 7]
```diff
@@ -62,45 +61,46 @@ object SourceDexcomPlugin : PluginBase(PluginDescription()
...
-            val glucoseValues = intent.getBundleExtra("glucoseValues")
-            for (i in 0 until glucoseValues.size()) {
-                val glucoseValue = glucoseValues.getBundle(i.toString())
-                val bgReading = BgReading()
-                bgReading.value = glucoseValue!!.getInt("glucoseValue").toDouble()
-                bgReading.direction = glucoseValue.getString("trendArrow")
-                bgReading.date = glucoseValue.getLong("timestamp") * 1000
-                bgReading.raw = 0.0
-                if (MainApp.getDbHelper().createIfNotExists(bgReading, "Dexcom$sensorType")) {
-                    if (SP.getBoolean(R.string.key_dexcomg5_nsupload, false)) {
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/source/SourceDexcomPlugin.kt]: missing. just to remember

File: app/src/main/java/info/nightscout/androidaps/plugins/source/SourceDexcomPlugin.kt [hunk 8]
```diff
@@ -62,45 +61,46 @@ object SourceDexcomPlugin : PluginBase(PluginDescription()
...
-                val timestamp = meter!!.getLong("timestamp") * 1000
-                if (MainApp.getDbHelper().getCareportalEventFromTimestamp(timestamp) != null) continue
-                val jsonObject = JSONObject()
-                jsonObject.put("enteredBy", "AndroidAPS-Dexcom$sensorType")
-                jsonObject.put("created_at", DateUtil.toISOString(timestamp))
-                jsonObject.put("eventType", CareportalEvent.BGCHECK)
-                jsonObject.put("glucoseType", "Finger")
-                jsonObject.put("glucose", meter.getInt("meterValue"))
-                jsonObject.put("units", Constants.MGDL)
-                NSUpload.uploadCareportalEntryToNS(jsonObject)
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/source/SourceDexcomPlugin.kt]: this is missing. jut to remember

File: app/src/main/java/info/nightscout/androidaps/plugins/source/SourceDexcomPlugin.kt [hunk 9]
```diff
@@ -62,45 +61,46 @@ object SourceDexcomPlugin : PluginBase(PluginDescription()
...
+                        meter.getInt("meterValue").toDouble()))
             }
-            if (SP.getBoolean(R.string.key_dexcom_lognssensorchange, false) && intent.hasExtra("sensorInsertionTime")) {
-                val sensorInsertionTime = intent.extras!!.getLong("sensorInsertionTime") * 1000
-                if (MainApp.getDbHelper().getCareportalEventFromTimestamp(sensorInsertionTime) == null) {
-                    val jsonObject = JSONObject()
-                    jsonObject.put("enteredBy", "AndroidAPS-Dexcom$sensorType")
-                    jsonObject.put("created_at", DateUtil.toISOString(sensorInsertionTime))
-                    jsonObject.put("eventType", CareportalEvent.SENSORCHANGE)
-                    NSUpload.uploadCareportalEntryToNS(jsonObject)
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/source/SourceDexcomPlugin.kt]: this is missing. jut to remember

File: app/src/main/java/info/nightscout/androidaps/plugins/treatments/fragments/TreatmentsBolusFragment.java [hunk 10]
```diff
@@ -248,7 +241,7 @@ public void onClick(View view) {
                         } else {
                             UploadQueue.removeID("dbAdd", _id);
                         }
-                        TreatmentsPlugin.getPlugin().getService().delete(treatment);
+                        //TreatmentsPlugin.getPlugin().getService().delete(treatment);
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/treatments/fragments/TreatmentsBolusFragment.java]: Not resolved

File: app/src/main/java/info/nightscout/androidaps/plugins/treatments/fragments/TreatmentsCareportalFragment.java [hunk 11]
```diff
@@ -179,7 +175,7 @@ public void onClick(View view) {
                 builder.setMessage(MainApp.gs(R.string.refresheventsfromnightscout) + " ?");
                 builder.setPositiveButton(MainApp.gs(R.string.ok), new DialogInterface.OnClickListener() {
                     public void onClick(DialogInterface dialog, int id) {
-                        MainApp.getDbHelper().resetCareportalEvents();
+                        //MainApp.getDbHelper().resetCareportalEvents();
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/treatments/fragments/TreatmentsCareportalFragment.java]: implement

File: app/src/main/java/info/nightscout/androidaps/plugins/treatments/fragments/TreatmentsCareportalFragment.java [hunk 12]
```diff
@@ -215,7 +211,7 @@ private void removeAndroidAPSStatedEvents() {
                 } else {
                     UploadQueue.removeID("dbAdd", _id);
                 }
-                MainApp.getDbHelper().delete(careportalEvent);
+                //MainApp.getDbHelper().delete(careportalEvent);
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/treatments/fragments/TreatmentsCareportalFragment.java]: implement

File: app/src/main/java/info/nightscout/androidaps/plugins/treatments/fragments/TreatmentsProfileSwitchFragment.java [hunk 13]
```diff
@@ -226,7 +216,7 @@ public void onClick(View view) {
                 builder.setMessage(MainApp.gs(R.string.refresheventsfromnightscout) + "?");
                 builder.setPositiveButton(MainApp.gs(R.string.ok), new DialogInterface.OnClickListener() {
                     public void onClick(DialogInterface dialog, int id) {
-                        MainApp.getDbHelper().resetProfileSwitch();
+                        //MainApp.getDbHelper().resetProfileSwitch();
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/treatments/fragments/TreatmentsProfileSwitchFragment.java]: implement

File: app/src/main/java/info/nightscout/androidaps/plugins/treatments/fragments/TreatmentsTempTargetFragment.java [hunk 14]
```diff
@@ -216,7 +208,7 @@ public void onClick(View view) {
                 builder.setMessage(MainApp.gs(R.string.refresheventsfromnightscout) + " ?");
                 builder.setPositiveButton(MainApp.gs(R.string.ok), new DialogInterface.OnClickListener() {
                     public void onClick(DialogInterface dialog, int id) {
-                        MainApp.getDbHelper().resetTempTargets();
+                        //MainApp.getDbHelper().resetTempTargets();
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/treatments/fragments/TreatmentsTempTargetFragment.java]: implement

File: app/src/main/java/info/nightscout/androidaps/services/DataService.java [hunk 15]
```diff
@@ -191,11 +191,11 @@ private void handleRemovedTreatmentFromNS(JSONObject json) {
         RxBus.INSTANCE.send(evtTreatment);
         // old DB model
         String _id = JsonHelper.safeGetString(json, "_id");
-        MainApp.getDbHelper().deleteTempTargetById(_id);
-        MainApp.getDbHelper().deleteTempBasalById(_id);
-        MainApp.getDbHelper().deleteExtendedBolusById(_id);
-        MainApp.getDbHelper().deleteCareportalEventById(_id);
-        MainApp.getDbHelper().deleteProfileSwitchById(_id);
+        //MainApp.getDbHelper().deleteTempTargetById(_id);
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/services/DataService.java]: implement

**Pull request closed without merging**